Content.Server/StationEvents/Events/MeteorSwarm.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Linq;
+using Content.Server.GameTicking;
+using Content.Server.Projectiles.Components;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+namespace Content.Server.StationEvents.Events
+{
+    public class MeteorSwarm : StationEvent
+    {
+        public override string Name => "MeteorSwarm";
+
+        public override int EarliestStart => 30;
+        public override float Weight => WeightLow;
+        public override int? MaxOccurrences => 2;
+        public override int MinimumPlayers => 20;
+
+        public override string StartAnnouncement => "Meteors are on a collision course with the station. Brace for impact.";
+        protected override string EndAnnouncement => "The meteor swarm has passed. Please return to your stations.";
+
+        public override string? StartAudio => "/Audio/Announcements/bloblarm.ogg";
+
+        protected override float StartAfter => 30f;
+        protected override float EndAfter => float.MaxValue;
+
+        private float _cooldown;
+
+        /// <summary>
+        /// We'll send a specific amount of waves of meteors towards the station per ending rather than using a timer.
+        /// </summary>
+        private int _waveCounter;
+
+        private const int MinimumWaves = 3;
+        private const int MaximumWaves = 8;
+
+        private const float MinimumCooldown = 10f;
+        private const float MaximumCooldown = 60f;
+
+        private const int MeteorsPerWave = 5;
+        private const float MeteorVelocity = 10f;
+        private const float MaxAngularVelocity = 0.25f;
+        private const float MinAngularVelocity = -0.25f;
+
+        public override void Startup()
+        {
+            base.Startup();
+            var robustRandom = IoCManager.Resolve<IRobustRandom>();
+            _waveCounter = robustRandom.Next(MinimumWaves, MaximumWaves);
+        }
+
+        public override void Shutdown()
+        {
+            base.Shutdown();
+            _waveCounter = 0;
+            _cooldown = 0f;
+            EndAfter = float.MaxValue;
+        }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            if (!Started) return;
+
+            if (_waveCounter <= 0)
+            {
+                EndAfter = float.MinValue;
+            }
```

Comment:
this seems to be missing a return statement?

Reply:
I must've accidentally nuked it when pushing